src/pymmcore_widgets/_mda/_channel_table_widget.py

```diff
@@ -34,32 +34,43 @@ class ChannelDict(TypedDict, total=False):
         acquire_every: int
 
 
-AlignCenter = Qt.AlignmentFlag.AlignHCenter | Qt.AlignmentFlag.AlignVCenter
-
-
 class ChannelTable(QGroupBox):
     """Widget providing options for setting up a multi-channel acquisition.
 
     The `value()` method returns a dictionary with the current state of the widget, in a
     format that matches one of the [useq-schema Channel
     specifications](https://pymmcore-plus.github.io/useq-schema/schema/axes/#useq.Channel).
+
+    Parameters
+    ----------
+    title : str
+        Title of the QGroupBox widget. Bt default, 'Channel'.
+    parent : QWidget | None
+        Optional parent widget. By default, None.
+    channel_group : str | None
+        Optional channel group that will be set as the widget's initial
+        ChannelGroup. By default, None.
+    mmcore : CMMCorePlus | None
+        Optional [`pymmcore_plus.CMMCorePlus`][] micromanager core.
+        By default, None. If not specified, the widget will use the active
+        (or create a new)
+        [`CMMCorePlus.instance`][pymmcore_plus.core._mmcore_plus.CMMCorePlus.instance].
     """
 
     valueChanged = Signal()
-    _channel_group: str
+    CH_GROUP_ROLE = 1
```

Comment:
1 is reserved for DecorationRole... see https://doc.qt.io/qt-6/qt.html#ItemDataRole-enum

use a UserRole plus some offset

Reply:
is it ok to do that?

`CH_GROUP_ROLE = Qt.UserRole + 1`

and then to set the item data:

`channel_combo.setItemData(0, _channel_group, self.CH_GROUP_ROLE)`

The index depends on the number of items in the combobox...